### ISSUE TITLE
Instances sporadically get a random username's timeline for RSS feeds

### ISSUE THREAD

**Issue by acarasimon96** [zedeus/nitter#428]
In #424, there have been reports of some Nitter instances returning RSS feeds of a random username (typically Japanese accounts) instead of the requested user. Unfortunately, those comments are beyond the scope of the first comment, although they are totally related to the same user timeline RSS endpoint, so I've decided to continue the discussion here to keep clean the issue mentioned earlier.

## Example observed behavior

The following is based on [this recent comment by me](https://github.com/zedeus/nitter/issues/424#issuecomment-901357852).

Say I add the RSS feed for [@loveliveevents](https://nitter.kavin.rocks/loveliveeevents) (an account whose tweets are posted in English) to my feed reader app on my PC. (I use Akregator on Manjaro KDE, if you're curious.) A few days later, I would suddenly get a notification saying I got 16 new items because the instance returned the most recent tweets from [some random Japanese user](https://nitter.kavin.rocks/nakiriayame) through the same [feed URL](https://nitter.kavin.rocks/loveliveevents). A few hours later, the instance gets the correct username's tweets again.

## Further details and where we're at so far

This bug has affected some, if not all, Nitter instances, particularly the [nitter.pussthecat.org](https://nitter.pussthecat.org) public instance. I've discovered this same exact behavior at [nitter.kavin.rocks](https://nitter.kavin.rocks) not long after creating this issue, but it went back to normal after [the instance owner restored service due to the server running low on storage space](https://github.com/zedeus/nitter/issues/424#issuecomment-901380315).

@PeterDaveHello has recently [set up a new Tor-only instance](https://github.com/zedeus/nitter/issues/424#issuecomment-900416176) to investigate this bug, and I will be on the lookout when my private instance or any of the public instances I use does the same. In the meantime, I'll be learning some Nim so that I can contribute code to this project.

CC @FireMasterK @TheFrenchGhosty @zedeus

**Comment by PeterDaveHello** [comment]
As @FireMasterK mentioned that the problem was caused by the disk out of space issue, I'll try to reproduce that scenario.

**Comment by TheFrenchGhosty** [comment]
>As @FireMasterK mentioned that the problem was caused by the disk out of space issue, I'll try to reproduce that scenario.

My server doesn't have any storage issue, it has hundreds of gigabytes free.

Nitter seems to "break" after a while, usually just restarting it fixes it, my instance currently restart when nitter crashes (so since it crashes around once a week, it should restart around once a week)

**Comment by PeterDaveHello** [comment]
@TheFrenchGhosty I got some result for #424 from the full disk, just hosted a new Nitter instance and use `dd` to full the disk, but the random RSS feed issue didn't come up yet.

**Comment by acarasimon96** [comment]
> This specific bug has been a thing for months

@TheFrenchGhosty I've been aware of this since I first saw this happen on your server with [the same username as in the example above](https://nitter.pussthecat.org/loveliveevents) sometime at the beginning of this year. I did not realize at the time that this has always existed since way before I discovered Nitter in the middle of 2020. That behavior led me to switch to another public instance and eventually start up my own. I still use some public instances today to lessen the risk of API request overloads on my instance, if there's such a thing after 669aa2c fixed many of the false alarms in that area.

**Comment by PeterDaveHello** [comment]
So for more than 3 days, tested Nitter on a disk full instance, got some error result for #424, but no random user RSS reproduced.

**Comment by zedeus** [comment]
It may be due to a bug in the Redis client library or the way it's used in my redpool library. Tweaking certain configs may exacerbate the issue to the point it becomes easily reproducible, but I don't have much time to do so. Specifically you could play around with `redisConnections` and `redisMaxConnections` in nitter.conf, it could theoretically happen when the pool reuses an already open connection, not sure. 

One thing I did observe is the RSS output from these incidents seems fairly consistent and correct, it's not quite two normal RSS feeds mashed together, maybe. If it is, that would very much point towards Redis, otherwise it's more likely to be related to the HTTP client pool. I was hoping the recent fix I did for the HTTP pool usage would solve this RSS issue, but alas that wasn't the case. It could be something totally different too, not sure, maybe some thorough checks throughout the code could try to identify the discrepancies in the output to help finding the cause, or even just provide a hint towards the right direction. One check could be something like this: compare the timeline profile id to non-retweet/quote tweets in the timeline. If even one tweet doesn't match, something might be wrong. Positives can be investigated closer, and from there the logging can be moved up the code paths to find where it all goes wrong. I may attempt this but I currently don't have much time, and my sysadmin that manages deployment of nitter.net is busy at the moment.

This is the last major bug affecting Nitter on a relatively wide scale, I really appreciate the help.

**Comment by acarasimon96** [comment]
I've just ran into this on my private instance now running my fork called [Nitter FT](https://github.com/acarasimon96/nitter-ft). The strange thing is that I was able to reproduce this on my phone with the proposed cookie-based home timeline feature (#363 is now merged into Nitter FT, by the way) but not the RSS endpoint.

At first, my instance returned only a single user's timeline showing only old tweets. When I unfollowed a few users, it suddenly returned a few random users that the instance has never fetched in addition to one of the expected ones. Strangely, clearing cookies on my browser (I use Firefox Beta from the Play Store) and following all the accounts one by one fixed it. What's more, I wasn't able to reproduce this at all on either my desktop or laptop right when the odd behavior happened.

Unfortunately, I couldn't find any logs both in stdout and on disk that pointed out any hints to what's caused that incident. But I think that this could be more of a problem with the HTTP pool or worse, a bug in the unofficial Twitter API.

I also find it ridiculous that logging hasn't been implemented well enough this late in the project. The lack of debug logging capabilities outside errors is what makes this and other annoying bugs in Nitter extremely hard to diagnose.
